state/relation.go

```diff
@@ -556,17 +583,17 @@ func (r *Relation) AllRemoteUnits(appName string) ([]*RelationUnit, error) {
 }
 
 // IsCrossModel returns whether this relation is a cross-model
-// relation.
-func (r *Relation) IsCrossModel() (bool, error) {
+// relation, and if so, the name of the remote application.
+func (r *Relation) IsCrossModel() (bool, *RemoteApplication, error) {
```

Comment:
Maybe it would be clearer to rename this to `RemoteApplication() (*RemoteApplication, error)` and have nil indicate that it's not cross-model? It doesn't seem to be used in many places. Or have both and implement `IsCrossModel()` in terms of `RemoteApplication()`?

Reply:
We need to retain the bool. I can rename to RemoteApplication()